File: mapbox_maps_flutter_mobile/android/src/main/kotlin/com/mapbox/maps/mapbox_maps/offline/OfflineSwitch.kt
package com.mapbox.maps.mapbox_maps.offline

import com.mapbox.maps.mapbox_maps.pigeons._OfflineSwitch

class OfflineSwitch : _OfflineSwitch {

  private val switcher = com.mapbox.common.OfflineSwitch.getInstance()

  override fun setMapboxStackConnected(connected: Boolean) {
    switcher.isMapboxStackConnected = connected
  }

  override fun isMapboxStackConnected(): Boolean {
    return switcher.isMapboxStackConnected
  }
}

File: mapbox_maps_flutter/lib/src/offline_switch.dart
import 'package:flutter/foundation.dart';
import 'package:mapbox_maps_flutter_platform_interface/mapbox_maps_flutter_platform_interface.dart';

/// Controls the Mapbox network stack connectivity.
class OfflineSwitch {
  final OfflineSwitchPlatformInterface _impl;

  @internal
  OfflineSwitch(this._impl);

  /// Returns whether the Mapbox network stack is connected.
  Future<bool> get isMapboxStackConnected => _impl.isMapboxStackConnected;

  /// Enables or disables the Mapbox network stack.
  Future<void> setMapboxStackConnected(bool isConnected) =>
      _impl.setMapboxStackConnected(isConnected);
}
